lib/nerves_network/application.ex

```diff
@@ -7,11 +7,19 @@ defmodule Nerves.Network.Application do
     import Supervisor.Spec, warn: false
 
     [resolvconf_file: resolvconf_file] = Application.get_env(:nerves_network, :resolver, [])
+    [ipv4: 
+     [
+       lease_file: _lf,
+       pid_file: _pf,
+       config_file: dhclientv4_config_file
+     ]
+    ] = Application.get_env(:nerves_network, :dhclientv4, [])
```

Comment:
Maybe use Application.fetch_env instead and use a case statement just incase some one was to forget to add / mess up the configuration. You could then log an error that the configuration was wrong, as opposed to crashing.

Reply:
Keyword lists aren't like Maps in that the order matters in the match.  We should probably look up the [:ipv4][config_file] key instead of matching as the user might have those options out of order (or they might have additional ones)